text/0074-source-context-via-links.md

```diff
@@ -0,0 +1,70 @@
+- Start Date: 2023-02-15
+- RFC Type: feature
+- RFC PR: <https://github.com/getsentry/rfcs/pull/74>
+- RFC Status: draft
+
+# Summary
+
+In situations where a source code link and a line number is available for a frame, present the link in the UI and show
+the source code surrounding the line.
+
+# Motivation
+
+The source context is currently composed of three [`Frame` attributes](https://develop.sentry.dev/sdk/event-payloads/stacktrace/#frame-attributes):
+
+- `context_line` - Source code in filename at `lineno`.
+- `pre_context` - A list of source code lines before `context_line` (in order) – usually [`lineno - 5:lineno`].
+- `post_context` - A list of source code lines after `context_line` (in order) – usually [`lineno + 1:lineno + 5`].
+
+Showing a source code context and a link to it (e.g. to GitHub) currently works only if the source code itself is
+available during symbolication, thus the attributes are filled in at that time, or if they were already sent with the event by the SDK.
+
+There are, however, situations where we do have a URL where the source code resides, but not the contents (without downloading it), for example:
+
+- Portable-PDB source-link (.NET, see [this `symbolic` issue](https://github.com/getsentry/symbolic/issues/735))
+- via a repository integration in combination with associated commit
+
+In these cases, the actual source code is not necessary to do symbolication (as opposed to source maps and other types of obfuscated source containers) but it is useful for end-user
+when evaluating the issue in the UI.
+
+## Related GH feature requests
+
+- [Extend stack-trace linking UI](https://github.com/getsentry/sentry/issues/35608)
+- [Support stack trace linking when stack frames do not contain line context.](https://github.com/getsentry/sentry/issues/44015)
+
+# Options Considered
+
+## A | Download the source code during symbolication
+
+One option is to download the source code from its link at the time the event is symbolicated, filling the context-related
+`Frame` attributes for each stacktrace frame, while the event is being processed.
+
+### Pros
+
+- Reuses existing fields (changes only needed in the `Symbolicator`).
+- Sources surrounding the frame line are then stored in the database, enabling [search-by-code](https://github.com/getsentry/sentry/issues/3755).
+- May be possible to do server-side authentication (based on project/org configuration) - this would complicate the solution & caching though, as opposed to only supporting publicly available sources.
+
+### Cons
+
+- Downloads all frame sources even though the event/frame may never be viewed by a user.

```

Comment:
Slightly related:
We should make "we want source context" configurable / an input option in symbolicator.
For example the profiling product goes through symbolicator, but does not use source context at all (they throw it away in the profiling processor AFAIK). We could potentially make profiles cheaper to symbolicate by just not doing source context resolution at all. CC @indragiek 

It may also be worth considering making the lines of context configurable, as it is hardcoded to `5` right now.